From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: cpw <cpw@weeksfamily.ca>
Date: Fri, 6 Nov 2015 15:16:23 -0500
Subject: [PATCH] Properly implement options for the renamer stuff. "JVN"
 argument enables JAD style variable naming (int i, j, k etc). Can pass a
 "RENAMER_FACTORY" to the decompiler context: this is the name of a class
 implementing IVariableNamingFactory, that will be loaded and instantiated,
 overriding other options.


diff --git a/src/org/jetbrains/java/decompiler/main/ClassWriter.java b/src/org/jetbrains/java/decompiler/main/ClassWriter.java
index c515d2e60e0917d2588a359c1d52a5345e8747ca..1338a42a36f02f479a7efbb280f2826a2948bb03 100644
--- a/src/org/jetbrains/java/decompiler/main/ClassWriter.java
+++ b/src/org/jetbrains/java/decompiler/main/ClassWriter.java
@@ -18,7 +18,6 @@ package org.jetbrains.java.decompiler.main;
 import org.jetbrains.java.decompiler.code.CodeConstants;
 import org.jetbrains.java.decompiler.main.ClassesProcessor.ClassNode;
 import org.jetbrains.java.decompiler.main.collectors.BytecodeMappingTracer;
-import org.jetbrains.java.decompiler.main.extern.IAbstractParameterRenamer;
 import org.jetbrains.java.decompiler.main.extern.IFernflowerLogger;
 import org.jetbrains.java.decompiler.main.extern.IFernflowerPreferences;
 import org.jetbrains.java.decompiler.main.rels.ClassWrapper;
diff --git a/src/org/jetbrains/java/decompiler/main/DecompilerContext.java b/src/org/jetbrains/java/decompiler/main/DecompilerContext.java
index 8d4d403773a861191083e2b2f6e469ec9ace45f2..d66145aef50938051b6a21f5d089b40cf9e8ef9f 100644
--- a/src/org/jetbrains/java/decompiler/main/DecompilerContext.java
+++ b/src/org/jetbrains/java/decompiler/main/DecompilerContext.java
@@ -68,11 +68,15 @@ public class DecompilerContext {
       try {
         currentContext.get().renamerFactory = Class.forName((String) DecompilerContext.getProperty(RENAMER_FACTORY)).asSubclass(IVariableNamingFactory.class).newInstance();
       } catch (Exception e) {
-
+        getLogger().writeMessage("Error loading renamer factory class", e);
       }
     }
     if (DecompilerContext.getNamingFactory() == null) {
-      currentContext.get().renamerFactory = new JADNameProvider.JADNameProviderFactory();
+      if (DecompilerContext.getOption(IFernflowerPreferences.USE_JAD_VARNAMING)) {
+        currentContext.get().renamerFactory = new JADNameProvider.JADNameProviderFactory();
+      } else {
+        currentContext.get().renamerFactory = new IdentityRenamerFactory();
+      }
     }
   }
 
diff --git a/src/org/jetbrains/java/decompiler/main/extern/IAbstractParameterRenamer.java b/src/org/jetbrains/java/decompiler/main/extern/IAbstractParameterRenamer.java
deleted file mode 100644
index 4c509eb6633f575b4f7cc93ad781885e8581be70..0000000000000000000000000000000000000000
--- a/src/org/jetbrains/java/decompiler/main/extern/IAbstractParameterRenamer.java
+++ /dev/null
@@ -1,7 +0,0 @@
-package org.jetbrains.java.decompiler.main.extern;
-
-import org.jetbrains.java.decompiler.main.rels.MethodWrapper;
-
-public interface IAbstractParameterRenamer {
-  public String renameParameter(String orig, int index, MethodWrapper wrapper, int flags);
-}
diff --git a/src/org/jetbrains/java/decompiler/main/extern/IFernflowerPreferences.java b/src/org/jetbrains/java/decompiler/main/extern/IFernflowerPreferences.java
index 79dd8e890cd37e88bfbe9f0c22c825b3c2549bb8..7940dcfbd89ec1427459aeeac1aebaea30db1899 100644
--- a/src/org/jetbrains/java/decompiler/main/extern/IFernflowerPreferences.java
+++ b/src/org/jetbrains/java/decompiler/main/extern/IFernflowerPreferences.java
@@ -62,6 +62,8 @@ public interface IFernflowerPreferences {
 
   String INCLUDE_ENTIRE_CLASSPATH = "iec";
 
+  String USE_JAD_VARNAMING = "jvn";
+
   Map<String, Object> DEFAULTS = Collections.unmodifiableMap(new HashMap<String, Object>() {{
     put(REMOVE_BRIDGE, "1");
     put(REMOVE_SYNTHETIC, "0");
@@ -96,5 +98,6 @@ public interface IFernflowerPreferences {
     put(UNIT_TEST_MODE, "0");
     put(DUMP_ORIGINAL_LINES, "0");
     put(INCLUDE_ENTIRE_CLASSPATH, "0");
+    put(USE_JAD_VARNAMING, "0");
   }});
 }
